tests/test_arcaflow_plugin_stressng.py

```diff
@@ -204,43 +206,41 @@ def test_functional_mq(self):
         self.assertEqual(res[1].mqinfo.stressor, "mq")
         self.assertGreaterEqual(math.ceil(res[1].mqinfo.wall_clock_time), 5)
 
-    # def test_functional_hdd(self):
-    #     hdd = stressng_schema.HDDStressorParams(
-    #         stressor="hdd",
-    #         workers=1,
-    #         hdd_bytes="100M",
-    #         hdd_opts=stressng_schema.HDDOptsParams(
-    #             [
-    #                 stressng_schema.HddOpts.DIRECT,
-    #                 stressng_schema.HddOpts.FSYNC,
-    #                 stressng_schema.HddOpts.WR_RND,
-    #             ]
-    #         ),
-    #         hdd_ops=10000,
-    #         hdd_write_size="4M",
-    #     )
-
-    #     stress = stressng_schema.StressNGParams(timeout="5s", stressors=[hdd])
-
-    #     reference_jobfile = "tests/reference_jobfile_hdd"
-
-    #     result = stress.to_jobfile()
-
-    #     for item in stress.stressors:
-    #         result = result + item.to_jobfile()
-
-    #     with open(reference_jobfile, "r") as file:
-    #         try:
-    #             reference = yaml.safe_load(file)
-    #         except yaml.YAMLError as e:
-    #             print(e)
-
-    #     self.assertEqual(yaml.safe_load(result), reference)
-    #     res = stressng_plugin.stressng_run(self.id(), stress)
-    #     print(res)
-    #     self.assertIn("success", res)
-    #     self.assertEqual(res[1].hddinfo.stressor, "hdd")
-    #     self.assertGreaterEqual(math.ceil(res[1].hddinfo.wall_clock_time), 5)
+    def test_functional_hdd(self):
+        hdd = stressng_schema.HDDStressorParams(
+            stressor="hdd",
+            workers=1,
+            hdd_bytes="100M",
+            hdd_opts=[
+                stressng_schema.HddOpts.DIRECT,
+                stressng_schema.HddOpts.FSYNC,
+                stressng_schema.HddOpts.WR_RND,
+            ],
+            hdd_ops=10000,
+            hdd_write_size="4M",
+        )
+
+        stress = stressng_schema.StressNGParams(timeout="5s", stressors=[hdd])
+
+        reference_jobfile = "tests/reference_jobfile_hdd"
+
+        result = stress.to_jobfile()
+
+        for item in stress.stressors:
+            result = result + item.to_jobfile()
+
+        with open(reference_jobfile, "r") as file:
+            try:
+                reference = yaml.safe_load(file)
+            except yaml.YAMLError as e:
+                print(e)
```

Comment:
Rather than `print(e)` (which has the side effect of eating the exception), how about `self.fail(e)`?

Or, frankly, just leave the exception unhandled:  it should never happen, and, if it does, it constitutes an error in the test framework (not a test failure)...and it'll produce a traceback which will include the stuff that the print would have output.

Ditto line 200 et al.